classes/Tools.php

```diff
@@ -2378,7 +2381,17 @@ public static function getMediaServer($filename)
             }
         }
 
-        if ($filename && self::$_cache_nb_media_servers && ($id_media_server = (abs(crc32($filename)) % self::$_cache_nb_media_servers + 1))) {
+        return self::$_cache_nb_media_servers > 0;
+    }
+
+    /**
+     * @param string $filename
+     *
+     * @return string
+     */
+    public static function getMediaServer($filename)
+    {
+        if ($filename && self::hasMediaServer() && ($id_media_server = (abs(crc32($filename)) % self::$_cache_nb_media_servers + 1))) {
```

Comment:
Otherwise you can't have a filename '0', which is a valid filename
```suggestion
        if (self::hasMediaServer() && ($id_media_server = (abs(crc32($filename)) % self::$_cache_nb_media_servers + 1))) {
```

Reply:
:white_check_mark: